File: proj.android_studio/ee-x-ads/src/main/java/com/ee/ads/ViewHelper.java
package com.ee.ads;

import android.graphics.Point;
import android.view.View;
import android.widget.FrameLayout;

import androidx.annotation.NonNull;

import com.ee.core.internal.Utils;

public class ViewHelper {
    private final View _view;

    public ViewHelper(@NonNull View view) {
        _view = view;
    }

    @NonNull
    public Point getPosition() {
        Utils.checkMainThread();
        /*
        int p[] = new int[2];
        view.getLocationInWindow(p);
        return new Point(p[0], p[1]);
        */
        FrameLayout.LayoutParams params = (FrameLayout.LayoutParams) _view.getLayoutParams();
        return new Point(params.leftMargin, params.topMargin);
    }

    public void setPosition(@NonNull Point position) {
        Utils.checkMainThread();
        FrameLayout.LayoutParams params = (FrameLayout.LayoutParams) _view.getLayoutParams();
        params.leftMargin = position.x;
        params.topMargin = position.y;
        _view.setLayoutParams(params);
    }

    @NonNull
    public Point getSize() {
        Utils.checkMainThread();
        // return new Point(view.getWidth(), view.getHeight());
        FrameLayout.LayoutParams params = (FrameLayout.LayoutParams) _view.getLayoutParams();
        return new Point(params.width, params.height);
    }

    public void setSize(@NonNull Point size) {
        Utils.checkMainThread();
        FrameLayout.LayoutParams params = (FrameLayout.LayoutParams) _view.getLayoutParams();
        params.width = size.x;
        params.height = size.y;
        _view.setLayoutParams(params);
    }

    public boolean isVisible() {
        Utils.checkMainThread();
        return _view.getVisibility() == View.VISIBLE;
    }

    public void setVisible(boolean visible) {
        Utils.checkMainThread();
        if (visible) {
            _view.setVisibility(View.VISIBLE);
        } else {
            // View.INVISIBLE doesn't trigger a view redraw.
            // Production: load and then call setVisible(true) doesn't show the native ad.
            _view.setVisibility(View.GONE);
        }
    }
}

File: proj.android_studio/ee-x-ads/src/main/java/com/ee/ads/ViewHelper.kt
package com.ee.ads

import android.graphics.Point
import android.view.View
import android.widget.FrameLayout
import androidx.annotation.UiThread
import com.ee.core.internal.Thread

class ViewHelper(private val _view: View) {
    var position: Point
        @UiThread get() {
            Thread.checkMainThread()
            /*
            int p[] = new int[2];
            view.getLocationInWindow(p);
            return new Point(p[0], p[1]);
            */
            val params = _view.layoutParams as FrameLayout.LayoutParams
            return Point(params.leftMargin, params.topMargin)
        }
        @UiThread set(value) {
            Thread.checkMainThread()
            val params = _view.layoutParams as FrameLayout.LayoutParams
            params.leftMargin = value.x
            params.topMargin = value.y
            _view.layoutParams = params
        }

    var size: Point
        @UiThread get() {
            Thread.checkMainThread()
            // return new Point(view.getWidth(), view.getHeight());
            val params = _view.layoutParams as FrameLayout.LayoutParams
            return Point(params.width, params.height)
        }
        @UiThread set(value) {
            Thread.checkMainThread()
            val params = _view.layoutParams as FrameLayout.LayoutParams
            params.width = value.x
            params.height = value.y
            _view.layoutParams = params
        }

    var isVisible: Boolean
        @UiThread get() {
            Thread.checkMainThread()
            return _view.visibility == View.VISIBLE
        }
        @UiThread set(value) {
            Thread.checkMainThread()
            if (value) {
                _view.visibility = View.VISIBLE
            } else {
                // View.INVISIBLE doesn't trigger a view redraw.
                // Production: load and then call setVisible(true) doesn't show the native ad.
                _view.visibility = View.GONE
            }
        }
}